example_contract_permissioned/README.md

```diff
@@ -0,0 +1,135 @@
+We define the USDC contract itself. It's a token contract where the token can only be bound/unbound from a UTXO if the address isn't on a blacklist
+
+```js
+token USDC {
+    abi {
+        // represents querying the MerkleTree contract to know if we can send USDC to a specific address
+        effect IsBlacklisted(addr: Address) -> UsdcTransferPermission::Intermediate
+    }
+
+    // can only bind (send to) this token to a UTXO owned by an address not on the blacklist
+    bind() {
+        const permission = raise IsBlacklisted(tx.context.caller);
+        const caller = raise Caller();
+        if (caller != permission.targetAddress) {
+            fail "Wrong address for transfer permission"
+        }
+        permission.burn();
+    }
+    // can only unbind (send from) this token to a UTXO owned by an address not on the blacklist
+    unbind() {
+        const permission = raise IsBlacklisted(tx.context.caller);
+        const caller = raise Caller();
+        if (caller != permission.targetAddress) {
+            fail "Wrong address for transfer permission"
+        }
+        permission.burn();
+    }
+}
+```
+
+```js
+token UsdcTransferPermission {
+    storage {
+        targetAddr: Address // keep track of who we're giving permission to send USDC to
+    }
+
+    mint(targetAddr: Address) {
+        const caller = raise Caller();
+        // only the USDC permission MerkleTree contract can mint permission tokens
+        if (caller !== UsdcPermission::address) {
+            fail "Transfer permission can only be minted by UsdcPermission contract"
+        }
+        return Intermediate { targetAddr }
+    }
+    burn() {}
+    bind() {
+        fail "Cannot bind UsdcTransferPermission token"
+    }
+}
+```
+
+```js
+
+utxo UsdcPermission {
+    abi {
+        // assume there exists some "MerkleTree" type implementation in Starstream
```

Comment:
We can remove this doc or remember to modify it later since this predates the linked list work